crates/apollo-compiler/src/database/hir.rs

```diff
@@ -2556,6 +2621,10 @@ impl UnionTypeDefinition {
             .add_extension(next_index, ext.union_members(), UnionMember::name);
         self.extensions.push(ext);
     }
+
+    pub fn implicit_field(&self) -> &FieldDefinition {
```

Comment:
Having a method named `implicit_field` (singular) feels weird to me. Why is it not plural and returning a slice? Alternatively, if this is always `__typename` and nothing else, why not call it `typename_implicit_field`?

Reply:
you're right, it is weird for it to be singular. let's do `implicit_fields` and returning a slice for all of them. we can try something else in the future if this doens't work out